### PR TITLE
Resolves #474: Add property support to UseDepVersion

### LINKED ISSUE #474: Add support for property defined version update in the goal use-dep-version
I noticed that currently `use-dep-version` works only when the version is defined inline, as in the following works:
```
mvn versions:use-dep-version  -DdepVersion=2.3.1 '-Dincludes=javax.xml.bind:jaxb-api'
```

```
--- a/example/pom.xml
+++ b/example/pom.xml
@@ -80,7 +80,7 @@
         <dependency>
             <groupId>javax.xml.bind</groupId>
             <artifactId>jaxb-api</artifactId>
-            <version>2.3.0</version>
+            <version>2.3.1</version>
         </dependency>
```

But the following property defined version doesn't get updated:
```
    <properties>
      <jaxb.version>2.3.0</jaxb.version>
    </properties>

        <dependency>
            <groupId>javax.xml.bind</groupId>
            <artifactId>jaxb-api</artifactId>
            <version>${jaxb.version}</version>
        </dependency>
```

This is a feature request to make `use-dep-version` work for property defined dependencies too.

Use case: we have many internal repositories which depends on few internal common libraries, sometimes we would want to update all of those repos to the latest version of the common libraries for bugs/security issues etc. But the downstream maven projects sometimes would have the version defined as a property or sometimes as inline version. If one command could handle both cases we could easily update all the repos to the target version.

### PR DESCRIPTION
**Pull request by ajarmoniuk**
- adding a new parameter, processProperties, which will also update property values if dependencies are controlled by properties
- the above parameter will be turned off in order to retain backwards compatibility

The behaviour of the property is described here:

```
/**
 * <p>Will augment normal processing by, if a dependency value is set using a property, trying to update
 * the value of the property.</p>
 * <p>If the property value is specified directly, will process it normally (as with {@code processProperties} being
 * {@code false}. If the property being updated is redefined in the reactor tree, will only change the property
 * value which lies closest to the dependency being updated. If the same property is also used to set
 * the value of another dependency, will not update that property value, and log a warning instead.
 * Finally, if the property value is specified in a parent file which is outside of the project, will log
 * a warning message.</p>
 * <p>Default is {@code false}.</p>
 *
 * @since 2.14.2
 */
```

also, the behaviour is also "described" in the unit tests.

@slawekjaranowski please review

By the way, I've put 2.14.2 tentatively, but because of the nature of this PR, I'd understand if it did not fit into the release.

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
Manual testing...

**Comment by ajarmoniuk** [comment]
Changed it so that it doesn't scan the whole tree if we're not processing properties.

Unfortunately, there's one problem I still don't know how to tackle -- maven unnecessarily descends into children even if the plugin could have made all necessary changes starting from root level. 

Do you know how to change that so that Maven doesn't run recursively -- from the plugin?

**Comment by slawekjaranowski** [comment]
I would like to postpone this for next minor release ... in next year 😄

**Comment by ajarmoniuk** [comment]
That's OK 😄

**Comment by ajarmoniuk** [comment]
So I guess that's a nice PR for 🎄😄

**Comment by ajarmoniuk** [comment]
Note to self: https://maven.apache.org/developers/mojo-api-specification.html#The_Descriptor_and_Annotations

Since there's time, I'll polish it a bit.

**Comment by ajarmoniuk** [comment]
Changed to aggregate mode; this change spurred the creation of a new enhancement: #880

**Comment by ajarmoniuk** [comment]
Temporarily converting to draft: let's see if I we can use maven (to do the recursion) and plugin context instead.

File: versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/UseDepVersionMojo.java [hunk 1]
```diff
@@ -62,12 +81,28 @@ public class UseDepVersionMojo extends AbstractVersionsDependencyUpdaterMojo {
...
+     * the value of the property.</p>
+     * <p>If the property value is specified directly, will process it normally (as with {@code processProperties} being
+     * {@code false}. If the property being updated is redefined in the reactor tree, will only change the property
+     * value which lies closest to the dependency being updated. If the same property is also used to set
+     * the value of another dependency, will not update that property value, and log a warning instead.
+     * Finally, if the property value is specified in a parent file which is outside of the project, will log
+     * a message.</p>
+     * <p>Default is {@code false}.</p>
+     *
+     * @since 2.14.2
```
Comment (slawekjaranowski) [inline versions-maven-plugin/src/main/java/org/codehaus/mojo/versions/UseDepVersionMojo.java:101]: I hope next version will be 2.15.0

**Pull request merged**